### PR TITLE
More RTD + cleanup

### PR DESCRIPTION
**Pull request by Gymnasiast**
The commits are somewhat related. It's recommended to look at them one by one.

### CODE REVIEW

File: src/openrct2/windows/_legacy.cpp [hunk 1]
```diff
@@ -449,29 +449,29 @@ void window_ride_construction_mouseup_demolish_next_piece(int32_t x, int32_t y,
...
         _currentTrackPieceDirection = direction & 3;
-        int32_t slope = _currentTrackCurve;
-        int32_t slopeEnd = _previousTrackSlopeEnd;
-        int32_t b2 = _currentTrackSlopeEnd;
-        int32_t bankEnd = _previousTrackBankEnd;
-        int32_t bankStart = _currentTrackBankEnd;
-        int32_t b5 = _currentTrackAlternative;
-        int32_t b4 = _currentTrackLiftHill;
+        int32_t savedCurrentTrackCurve = _currentTrackCurve;
+        int32_t savedPreviousTrackSlopeend = _previousTrackSlopeEnd;
```
Comment (Gymnasiast) [inline src/openrct2/windows/_legacy.cpp:453]: savedPreviousTrackSlopeEnd

File: src/openrct2/ride/Ride.h [hunk 2]
```diff
@@ -27,6 +27,7 @@ class StationObject;
 struct Peep;
 struct Ride;
 struct Staff;
+struct RideTypeDescriptor;
```
Comment (tupaschoal) [inline src/openrct2/ride/Ride.h:30]:
```suggestion
struct RideTypeDescriptor;
struct Staff;
```
Just to keep sorted.

File: src/openrct2/ride/Ride.cpp [hunk 3]
```diff
@@ -7821,3 +7783,20 @@ size_t Ride::FormatNameTo(void* argsV) const
...
+
+    return GetRideTypeDescriptor().RideModes;
+}
+
+const RideTypeDescriptor& Ride::GetRideTypeDescriptor() const
+{
+    if (type >= std::size(RideTypeDescriptors))
+    {
+        return DummyRTD;
+    }
```
Comment (tupaschoal) [inline src/openrct2/ride/Ride.cpp:7800]: Is this an expected scenario or a failure one? I'm just wondering if we should log an error inside this `if`.

File: src/openrct2/ride/RideData.cpp [hunk 4]
```diff
@@ -299,108 +299,6 @@ const rct_ride_name RideNaming[] =  {
...
-};
-
-const uint8_t AllRideModesAvailable[] = {
-    RIDE_MODE_CONTINUOUS_CIRCUIT, RIDE_MODE_CONTINUOUS_CIRCUIT_BLOCK_SECTIONED, RIDE_MODE_REVERSE_INCLINE_LAUNCHED_SHUTTLE, RIDE_MODE_POWERED_LAUNCH_PASSTROUGH, RIDE_MODE_SHUTTLE, RIDE_MODE_NORMAL, RIDE_MODE_BOAT_HIRE, RIDE_MODE_UPWARD_LAUNCH, RIDE_MODE_ROTATING_LIFT, RIDE_MODE_STATION_TO_STATION, RIDE_MODE_SINGLE_RIDE_PER_ADMISSION, RIDE_MODE_UNLIMITED_RIDES_PER_ADMISSION, RIDE_MODE_MAZE, RIDE_MODE_RACE, RIDE_MODE_BUMPERCAR, RIDE_MODE_SWING, RIDE_MODE_SHOP_STALL, RIDE_MODE_ROTATION, RIDE_MODE_FORWARD_ROTATION, RIDE_MODE_BACKWARD_ROTATION, RIDE_MODE_FILM_AVENGING_AVIATORS, RIDE_MODE_3D_FILM_MOUSE_TAILS, RIDE_MODE_SPACE_RINGS, RIDE_MODE_BEGINNERS, RIDE_MODE_LIM_POWERED_LAUNCH, RIDE_MODE_FILM_THRILL_RIDERS, RIDE_MODE_3D_FILM_STORM_CHASERS, RIDE_MODE_3D_FILM_SPACE_RAIDERS, RIDE_MODE_INTENSE, RIDE_MODE_BERSERK, RIDE_MODE_HAUNTED_HOUSE, RIDE_MODE_CIRCUS_SHOW, RIDE_MODE_DOWNWARD_LAUNCH, RIDE_MODE_CROOKED_HOUSE, RIDE_MODE_FREEFALL_DROP, RIDE_MODE_POWERED_LAUNCH, RIDE_MODE_POWERED_LAUNCH_BLOCK_SECTIONED, 0xFF
```
Comment (tupaschoal) [inline src/openrct2/ride/RideData.cpp:401]: What did the last `0xFF` on all lists meant? Just to stop iteration?

Comment (Gymnasiast) [reply src/openrct2/ride/RideData.cpp:401]: It was indeed to stop iteration.

**Pull request merged**